### PR TITLE
bjorncs/jrt transport metrics

### PR DESCRIPTION
**Pull request by bjorncs**
FYI @vekterli

### CODE REVIEW

File: jrt/src/com/yahoo/jrt/TransportMetrics.java [hunk 1]
```diff
@@ -76,4 +78,37 @@ public String toString() {
                 ", clientUnencryptedConnectionsEstablished=" + clientUnencryptedConnectionsEstablished +
                 '}';
     }
+
+    public static class Snapshot {
+        private final long tlsCertificateVerificationFailures, peerAuthorizationFailures, serverTlsConnectionsEstablished,
```
Comment (havardpe) [inline jrt/src/com/yahoo/jrt/TransportMetrics.java]: prefer untangling members

File: jrt/src/com/yahoo/jrt/TransportMetrics.java [hunk 2]
```diff
@@ -67,6 +73,15 @@ void incrementClientUnencryptedConnectionsEstablished() {
         clientUnencryptedConnectionsEstablished.incrementAndGet();
     }
 
+    void reset() {
```
Comment (havardpe) [inline jrt/src/com/yahoo/jrt/TransportMetrics.java]: maybe remove this?

File: jrt/tests/com/yahoo/jrt/EchoTest.java [hunk 3]
```diff
@@ -34,33 +35,35 @@
...
+                        (MetricsAssertions) metrics -> {
+                             assertEquals(1, metrics.serverTlsConnectionsEstablished());
+                             assertEquals(1, metrics.clientTlsConnectionsEstablished());
                         }}};
     }
 
     @Before
     public void setUp() throws ListenFailedException {
+        metrics =  TransportMetrics.getInstance();
+        metrics.reset();
```
Comment (havardpe) [inline jrt/tests/com/yahoo/jrt/EchoTest.java]: use snapshot instead?

File: container-disc/src/main/java/com/yahoo/container/jdisc/metric/JrtMetrics.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+/**
+ * Emits jrt metrics
+ *
+ * @author bjorncs
+ */
+class JrtMetrics {
+
+    private final TransportMetrics transportMetrics = TransportMetrics.getInstance();
+    private final Metric metric;
+    private volatile Snapshot previousSnapshot;
```
Comment (havardpe) [inline container-disc/src/main/java/com/yahoo/container/jdisc/metric/JrtMetrics.java]: either this should not need to be volatile or we need more synchronization elsewhere (the pattern is expected to be local cache for single writer)

**Review by havardpe — APPROVED** [review]
some minor comments

**Comment by bjorncs** [comment]
PTAL

**Review by havardpe — APPROVED** [review]
👍

**Comment by bjorncs** [comment]
Rebased with master

**Pull request merged**